machine_learning/loss_functions/perplexity_loss.py

```diff
@@ -0,0 +1,108 @@
+"""
+Compute the Perplexity which useful in predicting language model
+accuracy in Natural Language Processing (NLP.)
+Perplexity is measure of how certain the model in its predictions.
+
+Formula :
+
+Perplexity Loss = exp(-1/N (Σ ln(p(x)))
+
+Reference:
+https://en.wikipedia.org/wiki/Perplexity
+
+Author: Poojan Smart
+Email: smrtpoojan@gmail.com
+"""
+
+import numpy as np
+
+
+def perplexity_loss(y_true: np.ndarray, y_pred: np.ndarray) -> float:
+    """
+    Calculate the perplexity for the y_true and y_pred.
+
+    Args:
+        y_true: Actual label encoded sentences of shape (batch_size, sentence_length)
+        y_pred: Predicted sentences of shape (batch_size, sentence_length, vocab_size)
+
+    Returns:
+        Perplexity loss between y_true and y_pred.
+
+    >>> y_true = np.array([[1, 4], [2, 3]])
+    >>> y_pred = np.array(                 \
+        [[[0.28, 0.19, 0.21 , 0.15, 0.15], \
+          [0.24, 0.19, 0.09, 0.18, 0.27]], \
+          [[0.03, 0.26, 0.21, 0.18, 0.30], \
+           [0.28, 0.10, 0.33, 0.15, 0.12]]]\
+    )
```

Comment:
PEP8: Backslash line continuation should be avoided in Python.
```suggestion
    >>> y_pred = np.array(
    ...     [[[0.28, 0.19, 0.21 , 0.15, 0.15],
    ...       [0.24, 0.19, 0.09, 0.18, 0.27]],
    ...       [[0.03, 0.26, 0.21, 0.18, 0.30],
    ...       [0.28, 0.10, 0.33, 0.15, 0.12]]],
    ... )
```